linkshell_service/linkshell_service.lua

```diff
@@ -0,0 +1,76 @@
+local packets = require('packets')
+local shared = require('shared')
+
+local linkshell_data = shared.new('linkshell')
+
+-- Does this file need this?
+local linkshell.env = {
+    next = next,
+}
+  
+local linkshell = {
+    [1] = { message = { }, color = { }, },
+    [2] = { message = { }, },
+}
+
+handle_color = function(p)
+    local data = linkshell[1]
+        data.color.red = p.linkshell_red
+        data.color.green = p.linkshell_green
+        data.color.blue = p.linkshell_blue
+end
+
+packets.incoming[0x0CC]:register(function(p)
+    local ls_number = bit.band(p.flags, 0x40) == 0x40 and 2 or 1
+    local data = linkshell[ls_number]
+    data.name = p.linkshell_name
+    data.message.timestamp = p.timestamp
+    data.message.author = p.player_name
+    data.message.text = p.message
+    data.message.permissions = p.permissions
+end)
+
+packets.incoming[0x0E0]:register(function(p)
+    local data = linkshell[p.linkshell_number]
+    data.bag_index = p.bag_index
+end)
+
+packets.incoming[0x00D]:register(handle_color)
+packets.incoming[0x037]:register(handle_color)
+packets.incoming[0x0C9]:register(handle_color)
+
+local last_00D = packets.incoming[0x00D].last
+local last_037 = packets.incoming[0x037].last
+local last_0C9 = packets.incoming[0x0C9].last
+
+local handler_color(last_00D)
```

Comment:
Does this even load? The `local` here is incorrect.

Also, you should check if the packets aren't `nil` first, because they might be and the handler can't deal with that.